test/rubygems/test_gem_resolver.rb

```diff
@@ -266,14 +266,14 @@ def test_resolve_remote_missing_dependency
 
     res = Gem::Resolver.new [a_dep], Gem::Resolver::IndexSet.new
 
-    e = assert_raise Gem::UnsatisfiableDepedencyError do
+    e = assert_raise Gem::UnsatisfiableDependencyError do
```

Comment:
Is it time to remove the misspelled class as well?

https://github.com/rubygems/rubygems/blob/5b910cc4a78827c0571252dd93776b559cef608a/lib/rubygems/exceptions.rb#L283-L286

Reply:
I guess, yeah!

Reply:
Or warn first?
```ruby
Gem.deprecate_constant :UnsatisfiableDepedencyError
```

Reply:
Even if it's a very old constant, I guess it's safest to deprecate it first. We can do it on a separate PR.